File: backend/snus.py
from dataclasses import dataclass

@dataclass
class Snus:
    name: str = ""
    description: str = ""
    rating: int | None = None
    nicotine_g: float | None = None
    nicotine_portion: float | None = None
    portion_g: float | None = None
    weight_g: float | None = None
    portions: int | None = None
    snustype: str = "other"
    brand: str | None = None
    image: bytes | None = None
    image_mime: str | None = None

File: frontend/src/snus.ts
export class LocationAmount {
  id: number | undefined;
  amount: number | undefined;
  constructor(id: number, amount: number){
    this.id = id;
    this.amount = amount;
  }
}

export class Snus {
    id: number | null = null;
    name = "";
    description = "";
    rating: number | string | null = null ;
    nicotine_g: number | null = null;
    nicotine_portion: number | null = null;
    portion_g: number | null = null;
    weight_g: number | null = null;
    portions: number | null = null;
    type = "";
    brand: string | null = null;
    locations: LocationAmount[] = [];
    thumbnail_base64: string | null = null;
    thumbnail_mime: string | null = null;
}
